core/modules/wpublish/AddChunkGroupCommand.cc

```diff
@@ -107,11 +107,14 @@ AddChunkGroupCommand::run() {
         std::string const resource = "/chk/" + database + "/" + std::to_string(_chunk);
 
         LOGS(_log, LOG_LVL_DEBUG, "AddChunkGroupCommand::run  adding the chunk resource: "
-             << resource);
+             << resource << " in DataContext=" << clusterManager->DataContext());
 
         try {
-            // Notify XRootD/cmsd
+            // Notify XRootD/cmsd and (depending on a mode) modify the provider's copy
+            // of the inventory.
             clusterManager->Added(resource.c_str());
+            if (clusterManager->DataContext())
+                providerServer->GetChunkInventory().add(database, _chunk);
```

Comment:
Add curly braces or do it in one line: https://developer.lsst.io/coding/cpp_style_guide.html#single-statement-if-else-for-or-while-statements-must-only-be-written-without-brackets-if-on-one-line

Reply:
fixed